pytest-embedded-nuttx/pytest_embedded_nuttx/dut.py

```diff
@@ -75,8 +75,78 @@ def write_and_return(self, data: str, timeout: int = 2) -> AnyStr:
         ans = self.expect(pexpect.TIMEOUT, timeout=timeout)
         return ans.rstrip().decode()
 
+    def reset_to_nsh(self, ready_prompt: str = PROMPT_NSH) -> None:
+        """
+        Resets the board and waits until the Nuttshell prompt appears.
+        Defaults to 'nsh>'.
+
+        Args:
+            ready_prompt (str): string on prompt that signals completion.
+
+        Returns:
+            None
+        """
+        if self.reset:
+            logging.info('Resetting board')
+            self.reset()
+        else:
+            logging.error('Resetting method not available')
+        self.expect(ready_prompt, timeout=self.PROMPT_TIMEOUT_S)
+
+
+class NuttxSerialDut(SerialDut, NuttxDut):
+    """
+    DUT class for serial ports connected to generic boards running NuttX
+    with NuttX RTOS.
+    """
+
+    def __init__(
+        self,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+    def reset(self) -> None:
+        """Reset the DUT by toggling the DTR line.
+
+        Args:
+            None.
+
+        Returns:
+            None.
+        """
```

Comment:
```suggestion
        """Reset the DUT by toggling the DTR line."""
```

to simplify it. (a few more occurrences)